internal/runners/activate/activate.go

```diff
@@ -89,10 +86,16 @@ func (r *Activate) run(params *ActivateParams, activatorLoop activationLoopFunc)
 	// on --replace, replace namespace and commit id in as.yaml
 	if params.ReplaceWith.IsValid() {
 		var err error
-		projectToUse, err = updateProject(projectToUse.Source(), params.ReplaceWith)
+		err = updateProjectFile(projectToUse.Source(), params.ReplaceWith)
 		if err != nil {
 			return locale.WrapError(err, "err_activate_replace_write", "Could not update the project file with new namespace.")
 		}
+		var fail *failures.Failure
+		// re-read the project file and return the updated project
+		projectToUse, fail = project.Parse(projectToUse.Source().Path())
+		if fail != nil {
+			return locale.WrapError(fail.ToError(), "err_activate_reload", "Could not reload project.")
+		}
```

Comment:
You shouldn't need this at all. We're reloading the project when it's changed. If this is necessary something in the project writing logic is broken.

Reply:
I updated it, such that the project gets updated with the `SetNamespace` and `SetCommitID` calls that were only on the `projectfile.Project` struct.  I believe it is better to do these modifications on that level, as we do not only want to change the project *file* representation but also the actual representation of that file that we are using in the lines below.